libflux/src/core/semantic/nodes.rs

```diff
@@ -941,195 +955,275 @@ impl BinaryExpr {
         let cons = match self.operator {
             // The following operators require both sides to be equal.
             ast::Operator::AdditionOperator => Constraints::from(vec![
-                Constraint::Equal(
-                    self.left.type_of(),
-                    self.right.type_of(),
-                    self.right.loc().clone(),
-                ),
-                Constraint::Equal(self.left.type_of(), self.typ.clone(), self.loc.clone()),
-                Constraint::Kind(self.typ.clone(), Kind::Addable, self.loc.clone()),
+                Constraint::Equal {
+                    exp: self.left.type_of(),
+                    act: self.right.type_of(),
+                    loc: self.right.loc().clone(),
```

Comment:
What makes the left hand of a binary expression expected and the right hand side the actual type?

Reply:
That decision was arbitrary. As the left hand side is seen first, it seems natural that the left hand type is what was expected.

Reply:
ok sounds good.